src/main/java/nl/knaw/dans/managedeposit/core/service/IngestPathMonitor.java

```diff
@@ -47,31 +46,31 @@ public IngestPathMonitor(List<Path> depositBoxesPaths, DepositStatusUpdater depo
     }
 
     private void startMonitors() throws Exception {
-        IOFileFilter directories = FileFilterUtils.and(FileFilterUtils.directoryFileFilter(), HiddenFileFilter.VISIBLE);
-        IOFileFilter files = FileFilterUtils.and(FileFilterUtils.fileFileFilter(), FileFilterUtils.nameFileFilter("deposit.properties", IOCase.INSENSITIVE));
-        IOFileFilter filter = FileFilterUtils.or(directories, files);
-
-        log.info("Starting 'IngestPathMonitor', file filter: deposit.properties");
+        log.info("Starting 'IngestPathMonitor', with file filter: deposit.properties");
 
+        List<FileAlterationObserver> observers = new ArrayList<>();
         for (Path folder : toMonitorPaths) {
-            FileAlterationObserver observer = new FileAlterationObserver(folder.toFile(), filter);
+            IOFileFilter directories = FileFilterUtils.and(FileFilterUtils.directoryFileFilter(),/*HiddenFileFilter.VISIBLE*/new DepthFileFilter(folder, 1));
+            IOFileFilter files = FileFilterUtils.and(FileFilterUtils.fileFileFilter(), FileFilterUtils.nameFileFilter("deposit.properties", IOCase.INSENSITIVE));
+            IOFileFilter filter = FileFilterUtils.or(directories, files);
 
+            FileAlterationObserver observer = new FileAlterationObserver(folder.toFile(), filter);
             observer.addListener(this);
+            observers.add(observer);
 
-            FileAlterationMonitor monitor = new FileAlterationMonitor(this.pollingInterval, observer);
-            fileAlterationMonitors.add(monitor);
-
-            monitor.start();
-            log.debug("'IngestPathMonitor' is going to monitor the folder '{}'", folder);
         }
+
+        FileAlterationMonitor monitor = new FileAlterationMonitor(this.pollingInterval, observers.toArray(new FileAlterationObserver[0]));
```

Comment:
I don´t get this detour.

Reply:
Is the following text an answer to this review point:
second parameter of `FileAlterationMonitor` must be an array. 
`observers.toArray(new FileAlterationObserver[0])`: the function `toArray` calculates the necessary memory to allocate based on  a `FileAlterationObserver`.